app/upgrades/v4/upgrade.go

```diff
@@ -0,0 +1,90 @@
+package v4
+
+import (
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	consensusparamkeeper "github.com/cosmos/cosmos-sdk/x/consensus/keeper"
+	consensustypes "github.com/cosmos/cosmos-sdk/x/consensus/types"
+	crisistypes "github.com/cosmos/cosmos-sdk/x/crisis/types"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	// Ethermint modules
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+	feemarkettypes "github.com/evmos/ethermint/x/feemarket/types"
+)
+
+func GetStoreUpgrades() *storetypes.StoreUpgrades {
+	storeUpgrades := storetypes.StoreUpgrades{
+		Added: []string{
+			consensustypes.ModuleName,
+			crisistypes.ModuleName,
+		},
+	}
+	return &storeUpgrades
+}
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v3
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	paramsKeeper paramskeeper.Keeper,
+	consensusKeeper consensusparamkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		logger := ctx.Logger().With("upgrade", UpgradeName)
+		// Set param key table for params module migration
+		for _, subspace := range paramsKeeper.GetSubspaces() {
```

Comment:
why not get them by name by using `paramsKeeper.GetSubspace(moduleName)` instead of using a switch?

Reply:
That would lead to repetative code with the following code being implemented for every module
```golang
authSubpace, ok := paramsKeeper.GetSubspace(authtypes.ModuleName)
if ok {
	keytable := authtypes.ParamKeyTable()
	if !authSubpace.HasKeyTable() {
		authSubpace.WithKeyTable(keytable)
	}
}
```